server/sonar-web/src/main/js/apps/coding-rules/components/RuleDetailsProfiles.tsx

```diff
@@ -0,0 +1,286 @@
+      <Tooltip overlay={translate('coding_rules.activation_severity')}>
+        <span>
+          <SeverityHelper severity={activation.severity} />
+        </span>
+      </Tooltip>
+      {parentActivation !== undefined &&
+        activation.severity !== parentActivation.severity && (
+          <div className="coding-rules-detail-quality-profile-inheritance">
+            {translate('coding_rules.original')} {translate('severity', parentActivation.severity)}
+          </div>
+        )}
+    </td>
+  );
+
+  renderParameter = (param: { key: string; value: string }, parentActivation?: RuleActivation) => {
+    const originalParam =
+      parentActivation && parentActivation.params.find(p => p.key === param.key);
+    const originalValue = originalParam && originalParam.value;
+
+    return (
+      <div className="coding-rules-detail-quality-profile-parameter" key={param.key}>
+        <span className="key">{param.key}</span>
+        <span className="sep">{': '}</span>
+        <span className="value" title={param.value}>
+          {param.value}
+        </span>
+        {parentActivation &&
+          param.value !== originalValue && (
+            <div className="coding-rules-detail-quality-profile-inheritance">
+              {translate('coding_rules.original')} <span className="value">{originalValue}</span>
+            </div>
+          )}
+      </div>
+    );
+  };
+
+  renderParameters = (activation: RuleActivation, parentActivation?: RuleActivation) => (
+    <td className="coding-rules-detail-quality-profile-parameters">
+      {activation.params.map(param => this.renderParameter(param, parentActivation))}
+    </td>
+  );
+
+  renderActions = (activation: RuleActivation, profile: Profile) => {
+    const canEdit = profile.actions && profile.actions.edit && !profile.isBuiltIn;
+    const { ruleDetails } = this.props;
+    const hasParent = activation.inherit !== RuleInheritance.NotInherited && profile.parentKey;
+    return (
+      <td className="coding-rules-detail-quality-profile-actions">
+        {canEdit && (
+          <>
+            {!ruleDetails.isTemplate && (
+              <ActivationModalButton
+                activation={activation}
+                buttonText={translate('change_verb')}
+                className="coding-rules-detail-quality-profile-change"
+                modalHeader={translate('coding_rules.change_details')}
+                onDone={this.handleActivate}
+                profiles={[profile]}
+                rule={ruleDetails}
+              />
+            )}
+            {hasParent ? (
+              activation.inherit === RuleInheritance.Overridden && (
+                <ConfirmButton
+                  confirmButtonText={translate('yes')}
+                  confirmData={profile.key}
+                  modalBody={translateWithParameters(
+                    'coding_rules.revert_to_parent_definition.confirm',
+                    profile.parentName!
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: Forbidden non null assertion ([typescript:S2966](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=typescript%3AS2966))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2Fstas%2Frules&issues=AWEzCb2WmckxG3vItSGR&open=AWEzCb2WmckxG3vItSGR)

[ISSUE_KEY]: AWEzCb2WmckxG3vItSGR